store/store.go

```diff
@@ -351,6 +351,26 @@ func (s *SnapUbuntuStoreRepository) getAllPurchases(channel string, auther Authe
 	return purchasesByID, nil
 }
 
+// getMustBuy determines if a snap requires a payment, based on if it is non-free and if the user has already bought it
+func getMustBuy(prices map[string]float64, purchases []*purchase) bool {
+	// if the snap is free, then it doesn't need purchasing
+	if len(prices) == 0 {
+		return false
+	}
+
+	// search through all the purchases for an app to see if there are any
+	// that are for the whole app, not in-app.
```

Comment:
The comment here is a bit confusing. "... for an app ... that are for the whole app .. not in-app". Suspect some of those should say "snap" instead of app, but not sure.